ctap2token/token.go

```diff
@@ -0,0 +1,375 @@
+type CredentialRpEntity struct {
+	// ID is valid domain string that identifies the WebAuthn Relying Party.
+	ID   string `cbor:"id,omitempty"`
+	Name string `cbor:"name,omitempty"`
+	Icon string `cbor:"icon,omitempty"`
+}
+
+// CredentialUserEntity describes the user account to which
+// the new public key credential will be associated at the RP
+type CredentialUserEntity struct {
+	ID          []byte `cbor:"id"`
+	Name        string `cbor:"name,omitempty"`
+	DisplayName string `cbor:"displayName,omitempty"`
+	Icon        string `cbor:"icon,omitempty"`
+}
+
+type CredentialParam struct {
+	Type CredentialType `cbor:"type"`
+	Alg  Alg            `cbor:"alg"`
+}
+
+var (
+	PublicKeyRS256 CredentialParam = CredentialParam{Type: PublicKey, Alg: RS256}
+	PublicKeyPS256 CredentialParam = CredentialParam{Type: PublicKey, Alg: PS256}
+	PublicKeyES256 CredentialParam = CredentialParam{Type: PublicKey, Alg: ES256}
+)
+
+// CredentialType defines the type of credential, as defined in https://www.w3.org/TR/webauthn/#credentialType
+type CredentialType string
+
+const (
+	PublicKey CredentialType = "public-key"
+)
+
+// Alg must be the value of one of the algorithms registered on
+// https://www.iana.org/assignments/cose/cose.xhtml#algorithms.
+type Alg int
+
+const (
+	RS256          Alg = -257 // RSASSA-PKCS1-v1_5 using SHA-256
+	PS256          Alg = -37  // RSASSA-PSS w/ SHA-256
+	ECDHES_HKDF256 Alg = -25  // ECDH-ES + HKDF-256
+	ES256          Alg = -7   // ECDSA w/ SHA-256
+)
+
+// CredentialDescriptor defines a credential returned by the authenticator,
+// as defined by https://www.w3.org/TR/webauthn/#credential-dictionary
+type CredentialDescriptor struct {
+	ID         []byte                   `cbor:"id"`
+	Type       CredentialType           `cbor:"type"`
+	Transports []AuthenticatorTransport `cbor:"transports"`
+}
+
+// AuthenticatorTransport defines hints as to how clients might communicate with a particular authenticator,
+// as defined by https://www.w3.org/TR/webauthn/#transport.
+type AuthenticatorTransport string
+
+const (
+	// USB indicates the respective authenticator can be contacted over removable USB.
+	USB AuthenticatorTransport = "usb"
+	// NFC indicates the respective authenticator can be contacted over Near Field Communication (NFC).
+	NFC AuthenticatorTransport = "nfc"
+	// BLE indicates the respective authenticator can be contacted over Bluetooth Smart (Bluetooth Low Energy / BLE).
+	BLE AuthenticatorTransport = "ble"
+	// Internal indicates the respective authenticator is contacted using a client device-specific transport.
+	Internal AuthenticatorTransport = "internal"
+)
+
+type AuthenticatorOptions struct {
+	ResidentKey      bool `cbor:"rk,omitempty"`
+	UserVerification bool `cbor:"uv,omitempty"`
+}
+
+type PinProtocolVersion uint
+
+const (
+	PinProtoV1 PinProtocolVersion = 1
+)
+
+// MakeCredentialResponse...
+// CTAP 2.1 defines Fmt=0x1 and AuthData=0x2 while CTAP 2.0 defines AuthData=0x1 and Fmt=0x2 for some reasons
```

Comment:
This appears to be a typo in the 2.0 spec. The `authenticatorMakeCredential_Response` table in both list fmt=0x01 and authData=0x02.